torch/csrc/jit/mobile/function.cpp

```diff
@@ -49,6 +49,15 @@ bool Function::append_operator(
     }
   }
 
+  if (model_version == 0x3L &&
+      model_version < caffe2::serialize::kProducedBytecodeVersion &&
```

Comment:
Following the PR note, is the idea that the _convolution PR will bump kProducedBytecodeVersion and make this check true? 

Should there also be a kMaxSupportedBytecodeVersion, or is that redundant with kMaxSupportedFileFormatVersion?

Reply:
(Note on the server side we have separate notions of produced / max supported because the produced version is dynamic based on the operations in the graph.)

Reply:
> Following the PR note, is the idea that the _convolution PR will bump kProducedBytecodeVersion and make this check true?
Yes.
> 
> Should there also be a kMaxSupportedBytecodeVersion, or is that redundant with kMaxSupportedFileFormatVersion or kProducedBytecodeVersion?
For this use case (adding default arg to _convolution), we have the new code (version 4) supports both the old (v3) and new (v4) models. Full-jit does not need to change version numbers for this kind of update? It does hurt for old code (v3) running new models without _convolution. For this PR we don't support this.

I'm wondering if it's helpful to have dynamic version, so that the old code supports the new models without _convolution?
That's my understanding of dynamic versioning. Correct me if I'm wrong. 

Later for example if _relu is updated. When serializing bytecode, we check:
if _relu in model: 
  kProducedBytecodeVersion = 5
else if _convolution in model:
  kProducedBytecodeVersion = 4
else:
  kProducedBytecodeVersion = 3

Pro: for some model without curtain ops, it could run with bigger range of code versions.
Con: with more and more updated symbols, with logic could be complicated.

How do you think, @ljk53 @dreiss @xcheng16 ?

Reply:
> I'm wondering if it's helpful to have dynamic version, so that the old code supports the new models without _convolution?
> That's my understanding of dynamic versioning. Correct me if I'm wrong.

Your understanding is correct.
 
> Later for example if _relu is updated. When serializing bytecode, we check:
> if _relu in model:
> kProducedBytecodeVersion = 5
> else if _convolution in model:
> kProducedBytecodeVersion = 4
> else:
> kProducedBytecodeVersion = 3
>

Yep.

> Pro: for some model without curtain ops, it could run with bigger range of code versions.
> Con: with more and more updated symbols, with logic could be complicated.

That's true. We decided to do it on server, but it may not be necessary for mobile.

Reply:
As discussed with @ljk53 , we may keep it as is for now. Later some FC may be supported from active service, or we could come back and re-visit the option of dynamic versioning.